eth/handler.go

```diff
@@ -307,6 +311,18 @@ func (pm *ProtocolManager) handle(p *peer) error {
 			}
 		}()
 	}
+
+	// If we have any explicit whitelist block hashes, request them
+	if whitelist := pm.whitelist; whitelist != nil {
```

Comment:
No reason to create this whole assign-the-validate check if you don't use a pointer. A map can be nil, and you can iterate a nil map (will be a noop).

Reply:
Done.